### ISSUE TITLE
Security - 2FA Security codes copied as white text and appear as a white text on notepad/email/docs with white background

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#28140]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open Expensify App and navigate to Settings -> Security -> Two Factor Authentication
2. Security Codes appears to be copied
3. Click "Copy" button
4. Open Google docs OR email OR Notepad
5. Paste copied codes
6. Codes pasted with white text color

## Expected Result:
Codes should paste as black text color because most of the editors have white background.

## Actual Result:
 Codes appear as white text color

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.73.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/63d77de7-5ce8-4188-b90c-0b7d634f3b0b

https://github.com/Expensify/App/assets/93399543/d46e96da-cdbc-4ee3-b1f2-ff84a14b1e18

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ufumerfarooq
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695373596523109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Security - 2FA Security codes copied as white text and appear as a white text on notepad/email/docs with white background
### What is the root cause of that problem?

We use `setString` to set the contents to clipboard, but the main issue relies on the `Clipboard` that got exported by react-native web which does create a new empty node and add a styling to it and running copy. 

basically because of `Clipboard` coming from `react-native-web` is creating HTML even for small string which wrong and not good way to copy the content.

We have `navigator.clipboard` api in modern browser which we can use.

### What changes do you think we should make in order to solve the problem?

1. We should raise an upstream bug and resolve it there.

2. Update the clipboard file's code to use `navigator.clipboard` with checking the availability of backward compatibility 

3. Upgrade `@react-native-community/clipboard` package which now has support for web as well.

> package got renamed to `@react-native-clipboard/clipboard`

https://github.com/react-native-clipboard/clipboard/blob/master/src/Clipboard.web.ts which supports backward comparability for browsers as well.



I do recommend third option which is reliable and perfect fix.

### What alternative solutions did you explore? (Optional)
NA

**Comment by CortneyOfstad** [comment]
@michaelhaxhiu there is a known issue with duplicate assignments taking place via the `external` label for BZ — so removing my assignment 👍

**Comment by ufumerfarooq** [comment]
Contributor details
Your Expensify account email: umerfarooq@ultracodes.tech
Upwork Profile Link: https://www.upwork.com/freelancers/~01f33604ffac4199da

**Comment by ufumerfarooq** [comment]
Contributor details
Your Expensify account email: [umerfarooq@ultracodes.tech](mailto:umerfarooq@ultracodes.tech)
Upwork Profile Link: https://www.upwork.com/freelancers/~01f33604ffac4199da

**Comment by ufumerfarooq** [comment]
Contributor details
Your Expensify account email: [umerfarooq@ultracodes.tech](mailto:umerfarooq@ultracodes.tech)
Upwork Profile Link: https://www.upwork.com/freelancers/~01f33604ffac4199da

**Comment by ufumerfarooq** [comment]
Contributor details
Your Expensify account email: umerfarooq@ultracodes.tech
Upwork Profile Link: https://www.upwork.com/freelancers/~01f33604ffac4199da

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this to another BZ as part of my preparation for Sabbatical (starting Friday).

**Next steps:**
- Bug is reproduced and I agree with fixing it. It's small but it's weird, and this kind of weirdness can waiver user confidence in account security.
- @robertKozik is the C+ and will review the 2 proposals provided above

**Comment by NicMendonca** [comment]
@robertKozik Proposals for you here! ^

**Comment by NicMendonca** [comment]
@robertKozik bump bump ^

**Comment by NicMendonca** [comment]
@robertKozik bumped you here: https://expensify.slack.com/archives/C02NK2DQWUX/p1696542623775339

**Comment by robertKozik** [comment]
@b4s36t4 Are you sure that the web support from `react-native-community/clipboard` is official? I even though you linked the implementation, I cannot find any info about web support on npm page or github repo

**Comment by b4s36t4** [comment]
@react-native-community/clipboard is renamed to `react-native-clipboard/clipboard` this is the file https://github.com/react-native-clipboard/clipboard/blob/master/src/Clipboard.web.ts also https://github.com/react-native-clipboard/clipboard/blob/fecf673208f5ed67be4b1188ee69819f7518bd9f/src/Clipboard.web.ts#L25-L35 the implementation.

**Comment by robertKozik** [comment]
Yeah I saw that too, but It looks strange as all platforms are mentioned except web XD Did you check whether it's working after updating?

**Comment by b4s36t4** [comment]
No, I haven't checked. Do you want to me give it a go by upgrading?!

**Comment by b4s36t4** [comment]
I think since this is a migration from `@react-native-community/clipboard` they might've forget to add web support in readme.

**Comment by b4s36t4** [comment]
Sure, I'll check and update you in sometime.

**Comment by robertKozik** [comment]
Great, thanks!

**Comment by b4s36t4** [comment]
Hey @robertKozik , on replacing package to `react-native-clipboard/clipboard` its working fine on both mail.google.com and ms-word with `CMD+V` as shortcut  but in google docs should use `CMD+SHIFT+V`


https://github.com/Expensify/App/assets/59088937/05b86870-f9c0-4c22-a151-14502ede00cd


https://github.com/Expensify/App/assets/59088937/bf0f66f8-cb9b-4501-9c21-c60ab8362349


https://github.com/Expensify/App/assets/59088937/72a8a690-8b2d-4ff7-94fe-48e4778dab60

**Comment by robertKozik** [comment]
Thanks for checking it out. I think it should be way to go - updating and using the `react-native-clipboard/clipboard` library
from @b4s36t4 [Proposal](https://github.com/Expensify/App/issues/28140#issuecomment-1733864791)

🎀 👀 🎀 C+ reviewed

**Comment by ufumerfarooq** [comment]
I have already applied on this project and this is my upwork id:
https://www.upwork.com/freelancers/~01f33604ffac4199da

**Comment by b4s36t4** [comment]
@robertKozik @tgolen Raised PR here #29119

> Added mock for Clipboard API in `setup.js`

**Comment by NicMendonca** [comment]
@ufumerfarooq sent you offer!

@b4s36t4 can you please apply to the job: https://www.upwork.com/jobs/~0121601c821eb709ec

**Comment by ufumerfarooq** [comment]
Offer accepted

**Comment by b4s36t4** [comment]
@NicMendonca Accepted. Thanks :)

**Comment by NicMendonca** [comment]
sorry for the delay! everyone has now been paid. Thanks!!

**Comment by b4s36t4** [comment]
@NicMendonca I think this issue is eligible for Bonus.

I think there is some issue with melvin.

Issue Assigned: Oct 9
PR Merged: Oct 11

Can you please check.

**Comment by NicMendonca** [comment]
@b4s36t4 you got it!

<img width="595" alt="image" src="https://github.com/Expensify/App/assets/24947345/46c55fcc-18ed-4634-a310-ea1df8ec8488">
